spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -676,7 +677,6 @@ public boolean remove(ChannelInterceptor interceptor) {
 			}
 		}
 
-		@Nullable
 		public ChannelInterceptor remove(int index) {
 			ChannelInterceptor removed = this.interceptors.remove(index);
 			if (removed != null) {
```

Comment:
This one has to be removed as well.
That `this.interceptors.remove(index)` never returns null and may only throw an exception.
The confuse was really around this nullable variable.
If we remove this if altogether, all good.